### PR TITLE
BACKLOG-13954: handle lock and permission for disabling automatic lis…

### PR DESCRIPTION
**Pull request by jkevan**

### CODE REVIEW

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/ChildrenSection/AutomaticOrdering/AutomaticOrdering.spec.jsx [hunk 1]
```diff
@@ -69,8 +93,67 @@ describe('Automatic ordering component', () => {
...
+
+        const cmp = shallowWithTheme(
+            <AutomaticOrderingCmp {...props}/>,
+            {},
+            dsGenericTheme
+        );
+        expect(cmp.find('Button').props().disabled).toBe(true);
+    });
+
+    it('should disable add when node is locked', () => {
```
Comment (Taybou) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/ChildrenSection/AutomaticOrdering/AutomaticOrdering.spec.jsx:109]: should use different test name

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/ChildrenSection/AutomaticOrdering/AutomaticOrdering.spec.jsx [hunk 2]
```diff
@@ -69,8 +93,67 @@ describe('Automatic ordering component', () => {
...
+        const cmp = shallowWithTheme(
+            <AutomaticOrderingCmp {...props}/>,
+            {},
+            dsGenericTheme
+        );
+        expect(cmp.find('FieldContainer').at(1).props().inputContext.actionRender.props.disabled).toBe(true);
+        expect(cmp.find('FieldContainer').at(3).props().inputContext.actionRender.props.disabled).toBe(true);
+    });
+
+    it('should disable remove when node is locked', () => {
```
Comment (Taybou) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/ChildrenSection/AutomaticOrdering/AutomaticOrdering.spec.jsx:138]: same here

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/ChildrenSection/ChildrenSection.spec.jsx [hunk 3]
```diff
@@ -22,17 +42,36 @@ describe('Children section component', () => {
...
     });
 
+    it('should not be able to switch automatic ordering, if node is locked', () => {
+        context.nodeData.lockedAndCannotBeEdited = true;
+        setContext(context);
+        const cmp = shallowWithTheme(<ChildrenSectionCmp {...props}/>, {}, dsGenericTheme);
+        expect(cmp.find('WithStyles(ToggleCmp)').props().readOnly).toBe(true);
+    });
+
+    it('should not be able to switch automatic ordering, if node is locked', () => {
```
Comment (Taybou) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/ChildrenSection/ChildrenSection.spec.jsx:66]: same here

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/ChildrenSection/AutomaticOrdering/AutomaticOrdering.spec.jsx (hunk 1 above)
Comment (jkevan) [reply src/javascript/EditPanel/EditPanelContent/FormBuilder/ChildrenSection/AutomaticOrdering/AutomaticOrdering.spec.jsx]: done

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/ChildrenSection/AutomaticOrdering/AutomaticOrdering.spec.jsx (hunk 2 above)
Comment (jkevan) [reply src/javascript/EditPanel/EditPanelContent/FormBuilder/ChildrenSection/AutomaticOrdering/AutomaticOrdering.spec.jsx]: done

**Pull request merged**